src/hash_to_curve/chain.rs

```diff
@@ -326,6 +326,25 @@ pub fn chain_pm3div4(var0: &Fp) -> Fp {
 /// Bos-Coster (win=4) : 895 links, 17 variables
 /// Addition chain implementing exponentiation by (p**2 - 9) // 16.
 pub fn chain_p2m9div16(var0: &Fp2) -> Fp2 {
+    if cfg!(target_arch = "wasm32") {
```

Comment:
Maybe add a comment explaining why there is this special logic for wasm32.

Reply:
Done